<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ProcessingWidget</class>
 <widget class="QWidget" name="ProcessingWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1350</width>
    <height>850</height>
   </rect>
  </property>
  <layout class="QVBoxLayout" name="main_layout">
   <property name="spacing">
    <number>8</number>
   </property>
   <property name="leftMargin">
    <number>8</number>
   </property>
   <property name="topMargin">
    <number>8</number>
   </property>
   <property name="rightMargin">
    <number>8</number>
   </property>
   <property name="bottomMargin">
    <number>8</number>
   </property>
   <item>
    <widget class="QSplitter" name="main_splitter">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="childrenCollapsible">
      <bool>false</bool>
     </property>

     <!-- Left Panel: Processing Options -->
     <widget class="QWidget" name="options_panel">
      <property name="minimumSize">
       <size>
        <width>300</width>
        <height>0</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>400</width>
        <height>16777215</height>
       </size>
      </property>
      <layout class="QVBoxLayout" name="options_layout">
       <property name="spacing">
        <number>8</number>
       </property>
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>

       <!-- Load Data Section -->
       <item>
        <widget class="QGroupBox" name="load_group">
         <property name="title">
          <string>Load Photometry Data</string>
         </property>
         <layout class="QVBoxLayout" name="load_layout">
          <property name="spacing">
           <number>4</number>
          </property>
          <item>
           <layout class="QHBoxLayout" name="npz_row">
            <item>
             <widget class="QLineEdit" name="npz_path_edit">
              <property name="readOnly">
               <bool>true</bool>
              </property>
              <property name="placeholderText">
               <string>Select .npz file or use Tab 1 data...</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="btn_browse_npz">
              <property name="text">
               <string>...</string>
              </property>
              <property name="maximumSize">
               <size>
                <width>32</width>
                <height>16777215</height>
               </size>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QLabel" name="data_info_label">
            <property name="text">
             <string>No data loaded</string>
            </property>
            <property name="styleSheet">
             <string>color: #888888; font-size: 10px;</string>
            </property>
            <property name="wordWrap">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>

       <!-- dF/F Method Section -->
       <item>
        <widget class="QGroupBox" name="dff_group">
         <property name="title">
          <string>dF/F Calculation</string>
         </property>
         <layout class="QVBoxLayout" name="dff_layout">
          <property name="spacing">
           <number>4</number>
          </property>
          <item>
           <layout class="QHBoxLayout" name="method_row">
            <item>
             <widget class="QLabel" name="method_label">
              <property name="text">
               <string>Method:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="combo_dff_method">
              <item>
               <property name="text">
                <string>Fitted (regression)</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Simple (subtraction)</string>
               </property>
              </item>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QLabel" name="method_description">
            <property name="text">
             <string>Fitted: Linear regression of isosbestic to GCaMP for motion correction</string>
            </property>
            <property name="styleSheet">
             <string>color: #888888; font-size: 10px;</string>
            </property>
            <property name="wordWrap">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>

       <!-- Detrending Section -->
       <item>
        <widget class="QGroupBox" name="detrend_group">
         <property name="title">
          <string>Detrending</string>
         </property>
         <layout class="QVBoxLayout" name="detrend_layout">
          <property name="spacing">
           <number>4</number>
          </property>
          <item>
           <layout class="QHBoxLayout" name="detrend_method_row">
            <item>
             <widget class="QLabel" name="detrend_label">
              <property name="text">
               <string>Method:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="combo_detrend_method">
              <item>
               <property name="text">
                <string>None</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Linear</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Exponential</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Biexponential</string>
               </property>
              </item>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QCheckBox" name="chk_use_fit_range">
            <property name="text">
             <string>Use fit window for detrending</string>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="fit_range_row">
            <item>
             <widget class="QLabel" name="fit_start_label">
              <property name="text">
               <string>Start (min):</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QDoubleSpinBox" name="spin_fit_start">
              <property name="minimum">
               <double>0.000000000000000</double>
              </property>
              <property name="maximum">
               <double>1000.000000000000000</double>
              </property>
              <property name="singleStep">
               <double>0.500000000000000</double>
              </property>
              <property name="decimals">
               <number>1</number>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="fit_end_label">
              <property name="text">
               <string>End (min):</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QDoubleSpinBox" name="spin_fit_end">
              <property name="minimum">
               <double>0.000000000000000</double>
              </property>
              <property name="maximum">
               <double>1000.000000000000000</double>
              </property>
              <property name="singleStep">
               <double>0.500000000000000</double>
              </property>
              <property name="decimals">
               <number>1</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>

       <!-- Filtering Section -->
       <item>
        <widget class="QGroupBox" name="filter_group">
         <property name="title">
          <string>Filtering</string>
         </property>
         <layout class="QVBoxLayout" name="filter_layout">
          <property name="spacing">
           <number>4</number>
          </property>
          <item>
           <layout class="QHBoxLayout" name="lowpass_row">
            <item>
             <widget class="QCheckBox" name="chk_lowpass">
              <property name="text">
               <string>Low-pass filter:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QDoubleSpinBox" name="spin_lowpass_hz">
              <property name="minimum">
               <double>0.100000000000000</double>
              </property>
              <property name="maximum">
               <double>50.000000000000000</double>
              </property>
              <property name="value">
               <double>2.000000000000000</double>
              </property>
              <property name="suffix">
               <string> Hz</string>
              </property>
              <property name="enabled">
               <bool>false</bool>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="exclude_row">
            <item>
             <widget class="QLabel" name="exclude_label">
              <property name="text">
               <string>Exclude start:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QDoubleSpinBox" name="spin_exclude_start">
              <property name="minimum">
               <double>0.000000000000000</double>
              </property>
              <property name="maximum">
               <double>60.000000000000000</double>
              </property>
              <property name="singleStep">
               <double>0.500000000000000</double>
              </property>
              <property name="suffix">
               <string> min</string>
              </property>
              <property name="decimals">
               <number>1</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>

       <!-- Display Options -->
       <item>
        <widget class="QGroupBox" name="display_group">
         <property name="title">
          <string>Display Options</string>
         </property>
         <layout class="QVBoxLayout" name="display_layout">
          <property name="spacing">
           <number>4</number>
          </property>
          <item>
           <widget class="QCheckBox" name="chk_show_intermediates">
            <property name="text">
             <string>Show intermediate signals</string>
            </property>
            <property name="toolTip">
             <string>Show raw signals, fitted isosbestic, and detrend curve</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>

       <!-- Update Button -->
       <item>
        <widget class="QPushButton" name="btn_update_preview">
         <property name="text">
          <string>Update Preview</string>
         </property>
         <property name="styleSheet">
          <string>QPushButton { background-color: #0e639c; color: white; font-weight: bold; padding: 8px; border: none; border-radius: 3px; }
QPushButton:hover { background-color: #1177bb; }
QPushButton:pressed { background-color: #094771; }</string>
         </property>
        </widget>
       </item>

       <item>
        <spacer name="options_spacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>

       <!-- Fit Parameters Display -->
       <item>
        <widget class="QGroupBox" name="params_group">
         <property name="title">
          <string>Fit Parameters</string>
         </property>
         <layout class="QVBoxLayout" name="params_layout">
          <item>
           <widget class="QLabel" name="fit_params_label">
            <property name="text">
             <string>No fit performed yet</string>
            </property>
            <property name="styleSheet">
             <string>color: #888888; font-size: 10px; font-family: monospace;</string>
            </property>
            <property name="wordWrap">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>

     <!-- Right Panel: Preview -->
     <widget class="QWidget" name="preview_panel">
      <layout class="QVBoxLayout" name="preview_layout">
       <property name="spacing">
        <number>4</number>
       </property>
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <layout class="QHBoxLayout" name="preview_header_row">
         <item>
          <widget class="QLabel" name="preview_header">
           <property name="text">
            <string>Processed Signal Preview</string>
           </property>
           <property name="styleSheet">
            <string>font-size: 12px; font-weight: bold; color: #ffffff; padding: 4px 0px;</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="processing_label">
           <property name="text">
            <string></string>
           </property>
           <property name="styleSheet">
            <string>color: #ffcc00; font-style: italic;</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="preview_header_spacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QWidget" name="toolbar_container" native="true"/>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QWidget" name="canvas_container" native="true">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>400</height>
          </size>
         </property>
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
           <horstretch>1</horstretch>
           <verstretch>1</verstretch>
          </sizepolicy>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>

   <!-- Bottom Button Row -->
   <item>
    <layout class="QHBoxLayout" name="button_row">
     <property name="spacing">
      <number>8</number>
     </property>
     <item>
      <spacer name="button_spacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="btn_cancel">
       <property name="text">
        <string>Cancel</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btn_load_into_app">
       <property name="text">
        <string>Load into Application</string>
       </property>
       <property name="styleSheet">
        <string>QPushButton { background-color: #2ea043; color: white; font-weight: bold; padding: 8px 16px; border: none; border-radius: 3px; }
QPushButton:hover { background-color: #3fb950; }
QPushButton:pressed { background-color: #238636; }
QPushButton:disabled { background-color: #3e3e42; color: #888888; }</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>chk_lowpass</sender>
   <signal>toggled(bool)</signal>
   <receiver>spin_lowpass_hz</receiver>
   <slot>setEnabled(bool)</slot>
  </connection>
 </connections>
</ui>
